src/level_set/LevelSetUtilities.cpp

```diff
@@ -307,6 +309,66 @@ LevelSetUtilities::computeIntegralHeavisideFcns(LevelSetContainer* lsc)
 
 } // computeIntegralHeavisideFcns
 
+double
+LevelSetUtilities::computeNetInflowPhysicalBoundary(Pointer<HierarchyMathOps> hier_math_ops,
+                                                    int u_idx,
+                                                    int bdry_loc_idx)
+{
+    Pointer<PatchHierarchy<NDIM> > patch_hier = hier_math_ops->getPatchHierarchy();
+    const int wgt_sc_idx = hier_math_ops->getSideWeightPatchDescriptorIndex();
+    const int hier_finest_ln = patch_hier->getFinestLevelNumber();
+
+    double integral = 0.0;
+    for (int ln = 0; ln <= hier_finest_ln; ++ln)
+    {
+        Pointer<PatchLevel<NDIM> > patch_level = patch_hier->getPatchLevel(ln);
+        for (PatchLevel<NDIM>::Iterator p(patch_level); p; p++)
+        {
+            Pointer<Patch<NDIM> > patch = patch_level->getPatch(p());
+            Pointer<CartesianPatchGeometry<NDIM> > pgeom = patch->getPatchGeometry();
+            if (!pgeom->getTouchesRegularBoundary()) continue;
+
+            const tbox::Array<BoundaryBox<NDIM> > physical_codim1_boxes =
+                PhysicalBoundaryUtilities::getPhysicalBoundaryCodim1Boxes(*patch);
+            if (physical_codim1_boxes.size() == 0) continue;
+
+            Pointer<SideData<NDIM, double> > u_data = patch->getPatchData(u_idx);
+            Pointer<SideData<NDIM, double> > wgt_data = patch->getPatchData(wgt_sc_idx);
+
+            for (int n = 0; n < physical_codim1_boxes.size(); ++n)
+            {
+                const BoundaryBox<NDIM>& bdry_box = physical_codim1_boxes[n];
+                const unsigned int bdry_box_loc_idx = bdry_box.getLocationIndex();
+                if (bdry_box_loc_idx != bdry_loc_idx) continue;
+
+                const unsigned int axis = bdry_loc_idx / 2;
+                const unsigned int side = bdry_loc_idx % 2;
+                const bool is_lower = side == 0;
+                const double axis_normal = is_lower ? -1.0 : 1.0;
+
+                BoundaryBox<NDIM> trimmed_box = PhysicalBoundaryUtilities::trimBoundaryCodim1Box(bdry_box, *patch);
+                Box<NDIM> integration_box = PhysicalBoundaryUtilities::makeSideBoundaryCodim1Box(trimmed_box);
+
+                // Ensure the integration box is of width one in the normal direction
+                integration_box.upper(axis) = integration_box.lower(axis);
+
+                for (Box<NDIM>::Iterator b(integration_box); b; b++)
+                {
+                    const hier::Index<NDIM>& i = b();
+                    const SideIndex<NDIM> i_s(i, axis, SideIndex<NDIM>::Lower);
+
+                    const double u = (*u_data)(i_s);
+                    const double ds = (*wgt_data)(i_s);
+                    integral += -u * ds * axis_normal;
+                }
+            }
+        }
+    }
+
+    return IBTK_MPI::sumReduction(integral);
+
+} // computeNetInFlowPhysicalBoundary
+
 void
```

Comment:
@boyceg, @drwells and @abarret: does this implementation look correct for computing net inflow through a physical boundary?

Reply:
Sorry for the slow response - I see that this is also here: https://github.com/IBAMR/IBAMR/discussions/1615.

This generally looks right and if we just need to do one side of the boundary then the problem is a lot easier than an arbitrary subregion. `IBFEInstrumentPanel` and `IBInstrumentPanel` technically have some code to compute these quantities but the implementations are either not great (`IBFEInstrumentPanel` uses a pizza mesh) or incomplete.

Can you add some tests? Its pretty easy to mess up indexing with side-centered data. That's the best way to know for sure.

Reply:
Thanks! We plan to add a test to check the implementation. 

BTW, should we move this function to some other utility class? As written, it does not need any level set information  (but this can be changed easily). We are assuming only a single phase is present that the boundary location, which is the most case even for the intended multiphase applications we have in mind.

Reply:
Could do something similar to `SpongeLayerForceFunction`, which lets you evaluate flow rates through different domain boundaries.